### PR TITLE
Move IssueUser code to separate file

### PR DESCRIPTION
**Pull request by ethantkoenig**
IMO, our code is cleaner and easier to test with small files (such as `models/issue_user.go`) that are limited in scope.

Also add unit tests for `models/issue_user.go`), and removes the unused function `NewIssueUsers(..)`.

### CODE REVIEW

File: models/issue_user.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+type IssueUser struct {
+	ID          int64 `xorm:"pk autoincr"`
+	UID         int64 `xorm:"INDEX"` // User ID.
+	IssueID     int64
+	IsRead      bool
+	IsAssigned  bool
+	IsMentioned bool
+}
+
+func newIssueUsers(e Engine, repo *Repository, issue *Issue) error {
```
Comment (ethantkoenig) [inline models/issue_user.go:21]: Changed `e` to `Engine` (was previously `*xorm.Session`). Similarly for `updateIssueUserByAssignee(..)`

**Comment by lunny** [comment]
build failed.

**Comment by lunny** [comment]
LGTM

File: models/issue_user_test.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+package models
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_newIssueUsers(t *testing.T) {
```
Comment (andreynering) [inline models/issue_user_test.go:13]: `Test_newIssueUsers`-> `TestNewIssueUsers`

**Comment by lunny** [comment]
let L-G-T-M work

**Comment by andreynering** [comment]
One comment, otherwise, LGTM

**Review by appleboy — APPROVED** [review]

**Pull request merged**